compiler/src/dotty/tools/dotc/util/SourceFile.scala

```diff
@@ -136,6 +136,17 @@ case class SourceFile(file: AbstractFile, content: Array[Char]) extends interfac
     col
   }
 
+  /** The padding of the column corresponding to `offset`, includes tabs */
+  def startColumnPadding(offset: Int): String = {
+    var idx = startOfLine(offset)
+    val pad = new StringBuffer("")
```

Comment:
Use StringBuilder instead to avoid the synchronization overhead of StringBuffer, also you can use the empty constructor: `new StringBuilder`